Fix #942 RX_BUFFER_FULL

battjt opened the pull request:
Fix heap size to 512 MB (from 16 MB growing to 256 M).
Use G1 garbage collector configured to a max stop-the-world even of 40 ms.
Log all GC activity.
No sleeping for the rp1210 polling thread.

gumbelmj commented on etc/launch4j.xml:
```diff
@@ -1,26 +1,31 @@
+		<initialHeapSize>512</initialHeapSize>
+		<maxHeapSize>512</maxHeapSize>
+		<opt>-XX:+UseG1GC</opt>
+		<opt>-XX:MaxGCPauseMillis=40</opt>
```
Where did `40`ms come from? 

According to [Oracle](https://docs.oracle.com/javase/7/docs/technotes/guides/vm/gc-ergonomics.html)
>A value that is too small will cause the system to spend an excessive amount of time doing garbage collection.

Is that indicating the gc will have to do more passes or does that mean we are more likely to be paused?

battjt committed:
(fix #942) RX_BUFFER_FULL
Log GC events to c:\temp\j1939-84-gc.txt
Break link from streams to items on stream close().
Remove random warnings.
Fix heap size to 512 MB (from 16 MB growing to 256 M).
Use G1 garbage collector configured to a max stop-the-world even of 40 ms.
No sleeping for the rp1210 polling thread.

battjt commented:
I did a lot of testing with the G1GC with a max pause of 40 ms.  The worst I saw was 4 ms.  The biggest change is removing a memory leak related to an incomplete BAM message.  It would result in keeping all packets in memory until we ran out of memory, then a long GC cycle requiring a stop the world event would free it up.
